code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -1388,9 +1396,9 @@
 			return
 		if(istype(O, /obj/effect/alien/egg))
```

Comment:
replace this code with a typecache.

put this somewhere above this proc
```
GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(
	/obj/item/clothing/mask/facehugger,
	/obj/effect/alien/egg,
	/obj/structure/mineral_door,
	/obj/effect/alien/resin,
	/obj/structure/bed/nest)))
```
then do a check for `if(is_type_in_typecache(O, no_sticky_resin)`